Assets/Scripts/GameState/LeaderboardData.cs

```diff
@@ -56,58 +75,54 @@ public static void SaveScores(Highscores scores)
         stream.Close();
     }
 
-    public static Highscores LoadScores()
+    /// <summary>
+    /// Load all saved scores
+    /// </summary>
+    /// <returns>Returns a ordered list of Highscore structs</returns>
+    public static List<Highscore> LoadScores()
     {
         if (File.Exists(path))
         {
             BinaryFormatter formatter = new BinaryFormatter();
             FileStream stream = new FileStream(path, FileMode.Open);
 
-            Highscores data = formatter.Deserialize(stream) as Highscores;
-            stream.Close();
+            List<Highscore> data;
 
-            if(data.name.Length < 10 || data.score.Length < 10)
+            // Check if the file is corrupted
+            try
             {
-                // Scores are missing, delete the data and reset
+                data = formatter.Deserialize(stream) as List<Highscore>;
+            }
+            catch
+            {
+                stream.Close();
                 File.Delete(path);
-                data = LoadScores();
+                Debug.LogWarning("Highscore data was corrupted, it has been replaced.");
+                return LoadScores();
             }
 
+            stream.Close();
+
             return data;
         }
         else
         {
+            List<Highscore> highscores = new List<Highscore>();
+            
             // No leaderboard exists create a default one
-            int[] defaultScores = new int[10]{
-            5000,
-            4500,
-            4000,
-            3500,
-            3000,
-            2500,
-            2000,
-            1500,
-            1000,
-            0};
-
-            string[] defaultNames = new string[10]
-            {
-            "The Archetype",
-            "Fuereoduriko",
-            "Dabble",
-            "Frisk",
-            "Jesper",
-            "Rodrigues",
-            "Zedd",
-            "Grønnmerke",
-            "KHTangent",
-            "Endie"
-            };
-
-            Highscores data = new Highscores(defaultScores, defaultNames);
-
-            SaveScores(data);
-            return data;
+            highscores.Add(new Highscore(5000, "The Archetype"));
+            highscores.Add(new Highscore(4500, "Fuereoduriko"));
+            highscores.Add(new Highscore(4000, "Dabble"));
+            highscores.Add(new Highscore(3500, "Frisk"));
+            highscores.Add(new Highscore(3000, "Jesper"));
+            highscores.Add(new Highscore(2500, "Rodrigues"));
+            highscores.Add(new Highscore(2000, "Zedd"));
+            highscores.Add(new Highscore(1500, "Grønnmerke"));
+            highscores.Add(new Highscore(1000, "KHTangent"));
+            highscores.Add(new Highscore(10, "Endie"));
```

Comment:
Use a collection initializer instead: 😄
```suggestion
            List<Highscore> highscores = new List<Highscore>
            {
                new Highscore(5000, "The Archetype"),
                new Highscore(4500, "Fuereoduriko"),
                new Highscore(4000, "Dabble"),
                new Highscore(3500, "Frisk"),
                new Highscore(3000, "Jesper"),
                new Highscore(2500, "Rodrigues"),
                new Highscore(2000, "Zedd"),
                new Highscore(1500, "Grønnmerke"),
                new Highscore(1000, "KHTangent"),
                new Highscore(10, "Endie"),
            };
```